src/core/logical_layers/renderers/BivariateRenderer.tsx

```diff
@@ -64,23 +64,15 @@ const convertFillColorToRGBA = (fillColor: FillColor, withTransparency = true):
     withTransparency ? ',' + fillColor.a : ''
   })`;
 
-export const getH3GeoByLatLng = (
-  lngLat: LngLat,
-  resolution: number,
-): GeoJSON.Geometry => {
-  const h3 = geoToH3(
-    lngLat.lat,
-    lngLat.lng,
-    Math.min(resolution, MAX_BIVARIATE_GENERATED_RESOLUTION),
-  );
-  const h3Boundary = h3ToGeoBoundary(h3, true);
+function getH3Polygon(h3Index: string): GeoJSON.Geometry {
+  const h3Boundary = h3ToGeoBoundary(h3Index, true);
```

Comment:
no, you're not supposed to generate it at all, just hover the original feature using maplibre's mechanisms.